tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -60,24 +60,35 @@ namespace TensileLite
             }
         } // namespace math
 
-        constexpr size_t num_iters_per_cta(
-            size_t BLK_M, size_t BLK_N, size_t BLK_K, size_t m, size_t n, size_t k, int g)
+        constexpr size_t num_iters_total(size_t output_tiles, size_t iters_per_tile)
         {
-            return math::safe_ceil_div(math::safe_ceil_div(m, BLK_M) * math::safe_ceil_div(n, BLK_N)
-                                           * math::safe_ceil_div(k, BLK_K),
-                                       g);
+            return output_tiles * iters_per_tile;
         }
 
-        constexpr size_t number_of_output_tiles(size_t BLK_M, size_t BLK_N, size_t m, size_t n)
+        constexpr size_t num_iters_per_tile(size_t BLK_K, size_t k)
+        {
+            return math::safe_ceil_div(k, BLK_K);
+        }
+
+        constexpr size_t num_iters_per_cta(size_t iters_total, int g)
+        {
+            return math::safe_ceil_div(iters_total, g);
+        }
+
+        constexpr size_t number_of_output_tiles(size_t BLK_M, size_t BLK_N, size_t m, size_t n, size_t batch)
         {
             size_t m_tiles = math::safe_ceil_div(m, BLK_M);
             size_t n_tiles = math::safe_ceil_div(n, BLK_N);
-            return m_tiles * n_tiles;
+            return m_tiles * n_tiles * batch;
         }
 
-        constexpr size_t num_fixup_peers(size_t BLK_K, size_t k, size_t iters_per_cta)
+        constexpr size_t num_fixup_peers(size_t BLK_K, size_t k, size_t iters_total, size_t iters_per_tile, size_t iters_per_cta, size_t g)
```

Comment:
Based on the comment above, do we need to update the signature of this function if its only relying on iters_per_tile and  iters_per_cta?

Reply:
Good call, posted an update